src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -253,32 +193,34 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
         this.fetchFeedItems();
     };
 
-    /**
-     * Updates a task in the state via the API.
-     *
-     * @param {Object} args - A subset of the task
-     * @return void
-     */
-    updateTask = ({ text, id }: { id: string, text: string }): void => {
-        const { file, api } = this.props;
-        api.getFeedAPI(false).updateTask(file, id, text, this.feedSuccessCallback, this.feedErrorCallback);
+    updateTask = (task: TaskUpdatePayload, onSuccess: ?Function, onError: ?Function): void => {
+        const { file, api, onTaskUpdate = noop } = this.props;
+        const errorCallback = (e, code) => {
+            if (onError) {
+                onError(e, code);
+            }
+            this.feedErrorCallback(e, code);
+        };
+        const successCallback = () => {
+            this.feedSuccessCallback();
+
+            if (onSuccess) {
+                onSuccess();
+            }
+
+            onTaskUpdate();
+        };
+
+        api.getFeedAPI(false).updateTaskNew(file, task, successCallback, errorCallback);
 
         // need to load the pending item
         this.fetchFeedItems();
     };
 
-    /**
-     * Updates the task assignment
-     *
-     * @param {string} taskId - ID of task to be updated
-     * @param {string} taskAssignmentId - Task assignment ID
-     * @param {TaskAssignmentStatus} status - New task assignment status
-     * @return void
-     */
     updateTaskAssignment = (taskId: string, taskAssignmentId: string, status: TaskAssignmentStatus): void => {
         const { file, api } = this.props;
 
-        api.getFeedAPI(false).updateTaskAssignment(
+        api.getFeedAPI(false).updateTaskCollaborator(
```

Comment:
Is this intentional? Is it addressing a bug?

Reply:
I think I see what happened now. It looks like this is how `tasksNew` already worked, which was very similar to the prior flow, except for this one line.

Reply:
Yep